service.py

```diff
@@ -134,15 +138,11 @@ def get_gfs_from_biospecimen(self, biospecimen_id):
         Returns the links of biospecimen
         """
         resp = requests.get(
-            self.api+'/biospecimens/'+biospecimen_id)
-        if resp.status_code != 200 and len(resp.json()['results']) != 1:
-            return 'Biospecimen does not exist'
-        row = resp.json()
-        resp = requests.get(
-            self.api +
-            row['_links']['biospecimen_genomic_files']+'&limit=100')
+            self.api+'/genomic-files?biospecimen_id='+biospecimen_id)
```

Comment:
You either need to loop through pages or at least increase the `&limit=` on this request. The default 10 first-page results is definitely not sufficient.

Reply:
yep Updated it